server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -897,6 +897,15 @@ public Map<String, DataStream> findDataStreams(String... concreteIndices) {
         return builder.build();
     }
 
+    /**
+     * Checks whether the provided index has a parent data streams.
+     */
+    public boolean hasParentDataStream(String indexName) {
```

Comment:
I think "parent" here is confusing, perhaps we can just call it `boolean isPartOfDataStream(String indexName)`?